src/routes/(app)/+layout.server.ts

```diff
@@ -203,7 +203,9 @@ export async function load(params) {
 			.filter((d) => d._id !== null)
 			.map((d) => [
 				d._id,
-				wholeDiscount !== undefined && wholeDiscount > d.discountPercent
+				wholeDiscount !== undefined &&
+				d.discountPercent !== undefined &&
+				wholeDiscount > d.discountPercent
```

Comment:
this is not correct, if there's a whole discount and no individual discount, it will fallback to `d.discountPercent` which is `undefined` in that case